synapse/models/files.py

```diff
@@ -556,6 +577,54 @@ def getModelDefs(self):
                     ('base:ext', ('str', {}), {'ro': True,
                         'doc': 'The file extension.'}),
                 )),
+
+                ('file:mime:macho:loadcmd', {}, (
+                    ('file', ('file:bytes', {}), {
+                        'doc': 'The Mach-O file containing the load command.'}),
+                    ('type', ('int', {'enums': s_l_macho.getLoadCmdTypes()}), {
+                        'doc': 'The type of the load command'}),
+                    ('size', ('int', {}), {
+                        'doc': 'The size of the load command structure in bytes.'}),
+                )),
+                ('file:mime:macho:version', {}, (
+                    ('loadcmd', ('file:mime:macho:loadcmd', {}), {
+                        'doc': 'The associated load command node.'}),
+                    ('version', ('str', {}), {
+                        'doc': 'The version of the Mach-O file encoded in an LC_VERSION load command'})
+                )),
+                ('file:mime:macho:uuid', {}, (
+                    ('loadcmd', ('file:mime:macho:loadcmd', {}), {
+                        'doc': 'The actual load command '}),
+                    ('uuid', ('str', {}), {
+                        'doc': 'The UUID of the Mach-O application (as defined in an LC_UUID load command)'}),
+                )),
+                ('file:mime:macho:segment', {}, (
+                    ('loadcmd', ('file:mime:macho:loadcmd', {}), {
+                        'doc': 'The actual load command node.'}),
+                    ('loadcmd:file', ('file:bytes', {}), {
+                        'doc': 'The Mach-O file the segment is located in'}),
+                    ('name', ('str', {}), {
+                        'doc': 'The name of the Mach-O segment.'}),
+                    ('memsize', ('int', {}), {
+                        'doc': 'The size of the segment in bytes, when resident in memory, according to the load command structure.'}),
+                    ('disksize', ('int', {}), {
+                        'doc': 'The size of the segment in bytes, when on disk, according to the load command structure.'}),
+                    ('sha256', ('hash:sha256', {}), {
+                        'doc': 'The sha256 hash of the bytes of the segment.'}),
+                )),
+                ('file:mime:macho:section', {}, (
+                    ('segment', ('file:mime:macho:segment', {}), {
+                        'doc': 'The Mach-O segment that contains this section.'}),
+                    ('name', ('str', {}), {
+                        'doc': 'Name of the section.'}),
+                    ('size', ('int', {}), {
```

Comment:
Is there an offset we should snag here too?

Reply:
Aye, it's trivial to grab the file offset and we can add it to the node, and kinda like with sections, the file offset has a bit more meaning in this load command versus others.